lib/compress/zstd_compress.c

```diff
@@ -2593,8 +2613,8 @@ static size_t ZSTD_compress_frameChunk (ZSTD_CCtx* cctx,
         if (ms->nextToUpdate < ms->window.lowLimit) ms->nextToUpdate = ms->window.lowLimit;
 
         {   size_t cSize;
-            int useTargetCBlockSize = ZSTD_useTargetCBlockSize(&cctx->appliedParams);
-            if (useTargetCBlockSize && enoughDstCapacityForNoCompressSuperBlocks) {
+            if (ZSTD_useTargetCBlockSize(&cctx->appliedParams)
+              && ZSTD_enoughSuperBlockDstCapacityRemaining(cctx, dstCapacity, srcSize)) {
```

Comment:
The test is now played only when superblock is active.

But it doesn't change the issue that 
it effectively disables superblock whenever `dstCapacity` is not large enough to handle the worst case scenario (noCompress superblocks).

This pre-emptive disabling seems too strong because :
- The disabling happens without notification nor explanation. That's fine if if fixes an exceptional scenario, not if this happens regularly.
- most users will select either `dstCapacity = ZSTD_compressBound(srcSize)`, or `dstCapacity = srcSize` 
- `compressBound()` is the recommended method, but it's unclear to me if it provides enough to meet the new condition related to superblock. If not, that means that many (if not most) users will trigger this disabling, all transparently.

I do believe this has too big an impact.

Disabling superblock is still a valid strategy, but only _after_ the algorithm get a chance to compress the data with superblock. Only when that fails does it make sense to trigger backup strategies.